sklearn/linear_model/_ridge.py

```diff
@@ -795,22 +806,72 @@ def _ridge_regression(
         )
 
     if solver == "svd":
-        if sparse.issparse(X):
+        if X_is_sparse:
             raise TypeError("SVD solver does not support sparse inputs currently")
-        coef = _solve_svd(X, y, alpha)
+        coef = _solve_svd(X, y, alpha, xp)
 
     if ravel:
-        # When y was passed as a 1d-array, we flatten the coefficients.
-        coef = coef.ravel()
+        coef = _ravel(coef)
+
+    coef = xp.asarray(coef)
 
     if return_n_iter and return_intercept:
-        return coef, n_iter, intercept
+        res = coef, n_iter, intercept
     elif return_intercept:
-        return coef, intercept
+        res = coef, intercept
     elif return_n_iter:
-        return coef, n_iter
+        res = coef, n_iter
     else:
-        return coef
+        res = coef
+
+    return (*res, solver) if return_solver else res
+
+
+def _solver_auto_set(solver, positive, return_intercept, is_sparse, xp):
+    if solver != "auto":
+        return solver
+
+    is_numpy_namespace = _is_numpy_namespace(xp)
+
+    auto_solver_np = _solver_auto_set_np(positive, return_intercept, is_sparse)
+    if is_numpy_namespace:
+        return auto_solver_np
+
+    if positive:
+        raise ValueError(
+            "The solvers that support positive fitting do not support "
+            f"Array API dispatch to namespace {xp.__name__}. Please "
+            "either disable Array API dispatch, or use a numpy-like "
+            "namespace, or set `positive=False`."
+        )
+
+    # At the moment, Array API dispatch only supports the "svd" solver.
+    solver = "svd"
+    if solver != auto_solver_np:
+        warnings.warn(
+            f"Using Array API dispatch to namespace {xp.__name__} with "
+            f"`solver='auto'`will result in using the solver '{solver}'. "
+            "Results might be different than when Array API dispatch is "
+            "disabled, or when a numpy-like namespace is used, in which case "
+            f"the preferred solver would be '{auto_solver_np}'. Set "
+            f"`solver='{solver}'` to suppress this warning."
+        )
+
+    return solver
+
+
+def _solver_auto_set_np(positive, return_intercept, is_sparse):
```

Comment:
As with the other version of this function I am not a great fan of the name. My guess is that this resolves the `"auto"` value under the assumption that we are using Numpy arrays. Maybe `resolve_solver_with_numpy`? `resolve_solver_for_numpy`? I think we can afford to write "numpy" instead of "np", and somehow combine it with what we choose for the function above.

Reply:
Done in e51d6e4.